src/execution_plan/optimizations/optimize_cartesian_product.c

```diff
@@ -0,0 +1,140 @@
+/*
+* Copyright 2018-2020 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "optimize_cartesian_product.h"
+#include "../ops/op_filter.h"
+#include "../ops/op_cartesian_product.h"
+#include "optimizations_util.h"
+
+#define NOT_RESOLVED -1
+
+// Tests to see if given filter can be re-positioned.
+static inline bool _applicableFilter(const FT_FilterNode *f) {
```

Comment:
We can actually support all filters in this optimization, so this whole function can be deleted. This check is required for ApplyJoin because of memory inconsistencies when ValueHashJoin was operating on graph entities rather than scalars (an issue that could be resolved either in memory management or by adding ops like NodeHashJoin). It's not a concern here, though, as we don't alter the workings of Filter or CartesianProduct ops.

Reply:
thanks
removed